src/app/main.rs

```diff
@@ -423,18 +423,29 @@ where
                     .push(limit::layer(MAX_IN_FLIGHT))
                     .push(strip_header::layer(super::DST_OVERRIDE_HEADER))
                     .push(router::layer(|req: &http::Request<_>| {
-                        let addr = super::http_request_authority_addr(req)
-                            .or_else(|_| super::http_request_host_addr(req))
-                            .or_else(|_| super::http_request_orig_dst_addr(req))
-                            .ok();
-                        debug!("outbound addr={:?}", addr);
+                        let addr = || {
+                            super::http_request_authority_addr(req)
+                                .or_else(|_| super::http_request_host_addr(req))
+                                .or_else(|_| super::http_request_orig_dst_addr(req))
+                                .ok()
+                        };
                         super::http_request_l5d_override_dst_addr(req)
                             .ok()
                             .map(|override_addr| {
-                                debug!("outbound dst-override={:?}", override_addr);
+                                if log_enabled!(log::Level::Debug) {
```

Comment:
@kleimkuhler turns out I mislead you here a bit --- I just checked the `log` crate's macros, and the formatting _is_ already guarded by a similar `if` statement. So I think it _is_ okay to remove the `if` and just say
```rust 
debug!(
    "outbound addr={:?}; dst-override={:?}",
    addr(),
    override_addr
);
```

My bad, sorry about that.

Reply:
Oh cool sounds good! Thanks for actually checking; it probably would have been good for me to do the same instead of just asking.

Reply:
Hah, it's definitely my fault! :P